### PR TITLE
OTK5 Math Updates

### PR DESCRIPTION
**Pull request by BoyBaykiller**
* Remove 1-to-1 mappings of `System.Math` in `MathHelper `
* Replace the old quakes invsqrt with `MathF.ReciprocalSqrtEstimate`
* Use Vector512 in Equals functions
* ~~Call LengthSquared instead of typing out the equation everywhere~~
* add SIMD Equals for Matrix3(d), Matrix2x3(d), Matrix3x2(d), Vector2d and Vector3d
* remove unnecessary `using System.Globalization;`
* fix wrong `Vector2d.SizeInBytes`
* replace all `var` for all Vectors and some for Matrixes with explicit type

### CODE REVIEW

**Comment by BoyBaykiller** [comment]
Done

File: src/OpenTK.Mathematics/Matrix/Matrix4d.cs [hunk 1]
```diff
@@ -1813,23 +1813,13 @@ public override bool Equals(object obj)
...
         {
-            Vector256<double> aRow0 = Vector256.LoadUnsafe(ref Row0.X);
-            Vector256<double> bRow0 = Vector256.LoadUnsafe(ref other.Row0.X);
+            Vector512<double> aRow01 = Vector512.LoadUnsafe(ref Row0.X);
+            Vector512<double> aRow23 = Vector512.LoadUnsafe(ref Row0.X);
 
-            Vector256<double> aRow1 = Vector256.LoadUnsafe(ref Row1.X);
-            Vector256<double> bRow1 = Vector256.LoadUnsafe(ref other.Row1.X);
+            Vector512<double> bRow01 = Vector512.LoadUnsafe(ref other.Row0.X);
+            Vector512<double> bRow23 = Vector512.LoadUnsafe(ref other.Row0.X);
```
Comment (NogginBops) [inline src/OpenTK.Mathematics/Matrix/Matrix4d.cs:1820]: Should be `Row2.X`

File: src/OpenTK.Mathematics/Matrix/Matrix4d.cs [hunk 2]
```diff
@@ -1813,23 +1813,13 @@ public override bool Equals(object obj)
         [Pure]
         public bool Equals(Matrix4d other)
         {
-            Vector256<double> aRow0 = Vector256.LoadUnsafe(ref Row0.X);
-            Vector256<double> bRow0 = Vector256.LoadUnsafe(ref other.Row0.X);
+            Vector512<double> aRow01 = Vector512.LoadUnsafe(ref Row0.X);
+            Vector512<double> aRow23 = Vector512.LoadUnsafe(ref Row0.X);
```
Comment (NogginBops) [inline src/OpenTK.Mathematics/Matrix/Matrix4d.cs:1817]: Should be `Row2.X`

File: src/OpenTK.Mathematics/Matrix/Matrix2x4d.cs [hunk 3]
```diff
@@ -825,13 +825,9 @@ public override bool Equals(object obj)
...
-            Vector256<double> aRow0 = Vector256.LoadUnsafe(ref Row0.X);
-            Vector256<double> bRow0 = Vector256.LoadUnsafe(ref other.Row0.X);
-
-            Vector256<double> aRow1 = Vector256.LoadUnsafe(ref Row1.X);
-            Vector256<double> bRow1 = Vector256.LoadUnsafe(ref other.Row1.X);
-
-            return aRow0 == bRow0 && aRow1 == bRow1;
+            Vector512<double> aRow01 = Vector512.LoadUnsafe(ref Row0.X);
+            Vector512<double> bRow01 = Vector512.LoadUnsafe(ref other.Row0.X);
+            return aRow01 == bRow01;
```
Comment (NogginBops) [inline src/OpenTK.Mathematics/Matrix/Matrix2x4d.cs:830]: To match the formatting of the other implementations I think we should have blank lines separating these.

File: src/OpenTK.Mathematics/Matrix/Matrix4.cs [hunk 4]
```diff
@@ -1961,13 +1961,10 @@ public override bool Equals(object obj)
...
-            Vector256<float> aRow01 = Vector256.LoadUnsafe(ref Row0.X);
-            Vector256<float> bRow01 = Vector256.LoadUnsafe(ref other.Row0.X);
+            Vector512<float> aRow0123 = Vector512.LoadUnsafe(ref Row0.X);
+            Vector512<float> bRow0123 = Vector512.LoadUnsafe(ref other.Row0.X);
 
-            Vector256<float> aRow23 = Vector256.LoadUnsafe(ref Row2.X);
-            Vector256<float> bRow23 = Vector256.LoadUnsafe(ref other.Row2.X);
-
-            return aRow01 == bRow01 && aRow23 == bRow23;
+            return aRow0123 == bRow0123;
```
Comment (NogginBops) [inline src/OpenTK.Mathematics/Matrix/Matrix4.cs:1967]: To match the formatting of the other implementations I think we should have blank lines separating these.

File: src/OpenTK.Mathematics/Matrix/Matrix4x3.cs [hunk 5]
```diff
@@ -1058,11 +1058,10 @@ public override bool Equals(object obj)
...
         {
             Vector256<float> aRow012xy = Vector256.LoadUnsafe(ref Row0.X);
-            Vector256<float> bRow012xy = Vector256.LoadUnsafe(ref other.Row0.X);
+            Vector128<float> aRow2z3 = Vector128.LoadUnsafe(ref Row2.Z); // Note that we ref .Z and not .X.
 
-            // Note that we ref .Z and not .X.
-            Vector128<float> aRow2z3 = Vector128.LoadUnsafe(ref Row2.Z);
-            Vector128<float> bRow2z3 = Vector128.LoadUnsafe(ref other.Row2.Z);
+            Vector256<float> bRow012xy = Vector256.LoadUnsafe(ref other.Row0.X);
+            Vector128<float> bRow2z3 = Vector128.LoadUnsafe(ref other.Row2.Z); // Note that we ref .Z and not .X.
```
Comment (NogginBops) [inline src/OpenTK.Mathematics/Matrix/Matrix4x3.cs:1064]: The comments should have their own line, we don't put them at the end of code.

File: src/OpenTK.Mathematics/Matrix/Matrix4x3d.cs [hunk 6]
```diff
@@ -1056,18 +1056,13 @@ public override bool Equals(object obj)
...
         {
-            Vector256<double> aRow01x = Vector256.LoadUnsafe(ref Row0.X);
-            Vector256<double> bRow01x = Vector256.LoadUnsafe(ref other.Row0.X);
-
-            // Note that we ref .Y and not .X.
-            Vector256<double> aRow1yz2xy = Vector256.LoadUnsafe(ref Row1.Y);
-            Vector256<double> bRow1yz2xy = Vector256.LoadUnsafe(ref other.Row1.Y);
-
-            // Note that we ref .Z and not .X.
+            Vector512<double> aRow01xy = Vector512.LoadUnsafe(ref Row0.X);
```
Comment (NogginBops) [inline src/OpenTK.Mathematics/Matrix/Matrix4x3d.cs:1059]: This should be `aRow012xy`

File: src/OpenTK.Mathematics/Matrix/Matrix4x3d.cs [hunk 7]
```diff
@@ -1056,18 +1056,13 @@ public override bool Equals(object obj)
...
             Vector256<double> aRow2z3 = Vector256.LoadUnsafe(ref Row2.Z);
+
+            Vector512<double> bRow01xy = Vector512.LoadUnsafe(ref other.Row0.X);
```
Comment (NogginBops) [inline src/OpenTK.Mathematics/Matrix/Matrix4x3d.cs:1062]: This should be `bRow012xy`

File: src/OpenTK.Mathematics/Vector/Vector2.cs [hunk 8]
```diff
@@ -174,7 +172,7 @@ public void Normalize()
         /// </summary>
         public void NormalizeFast()
         {
-            var scale = MathHelper.InverseSqrtFast((X * X) + (Y * Y));
+            var scale = MathF.ReciprocalSqrtEstimate((X * X) + (Y * Y));
```
Comment (NogginBops) [inline src/OpenTK.Mathematics/Vector/Vector2.cs:175]: We should remove a few `var`s while we are at it I think.

**Review by NogginBops — CHANGES_REQUESTED** [review]

**Comment by BoyBaykiller** [comment]
You can wait with your review. I'll do one more commit

**Review by NogginBops — APPROVED** [review]

**Comment by NogginBops** [comment]
Merging.

**Pull request merged**